src/RedisMessaging.ReliableDelivery/Subscribe/ReliableSubscriber.cs

```diff
@@ -14,32 +16,35 @@ namespace RedisMessaging.ReliableDelivery.Subscribe
     /// </summary>
     public class ReliableSubscriber : IReliableSubscriber
     {
-        private static readonly char[] Separator = { ReliablePublisher.MessagePartSeparator };
-        private readonly IMessageValidator _messageValidator;
-        private readonly IMessageValidationFailureHandler _messageValidationFailureHandler;
+        private readonly IMessageParser _messageParser;
         private readonly ILogger<ReliableSubscriber> _log;
-        private readonly StackExchange.Redis.ISubscriber _subscriber;
+        private readonly IConnectionMultiplexer _connectionMultiplexer;
         private readonly ConcurrentDictionary<string, ChannelMessageQueue> _queues = new ConcurrentDictionary<string, ChannelMessageQueue>();
 
         public ReliableSubscriber(
             IConnectionMultiplexer connectionMultiplexer,
-            IMessageValidator messageValidator,
-            IMessageValidationFailureHandler messageValidationFailureHandler,
-            ILogger<ReliableSubscriber> log)
+            IMessageParser messageParser,
+            ILogger<ReliableSubscriber> log = null)
         {
-            _messageValidator = messageValidator;
-            _messageValidationFailureHandler = messageValidationFailureHandler;
-            _log = log;
-            _subscriber = connectionMultiplexer.GetSubscriber();
+            _connectionMultiplexer = connectionMultiplexer;
+            _messageParser = messageParser;
+            _log = log ?? NullLogger<ReliableSubscriber>.Instance;
         }
 
-        public void Subscribe(string channel, Action<string, string> handler)
+        protected virtual ISubscriber Subscriber => _connectionMultiplexer.GetSubscriber();
+
+        // for testing
+        internal Exception LastException { get; private set; }
+        internal long ExceptionsCount { get; private set; }
+        

```

Comment:
I don't like the idea of adding any properties for testing purposes only. As you can't check exceptions by asserts here I would prefer e.g. providing a custom testing logger and examining  the content of it or making the `HandleMessage` function virtual and overriding for testing purposes only.

Reply:
Fixed in commit c0e6a6a5a318baa8665bc086ba58fd7a66c6fd04